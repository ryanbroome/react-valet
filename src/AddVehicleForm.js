import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import ThemeContext from "./ThemeContext";

// add method either passed as props or saved in context ?
const AddVehicleForm = () => {
  const { addVehicle } = useContext(ThemeContext);
  const history = useHistory();

  const INITIAL_STATE = {
    ticketNum: "",
    // vehicleStatus: "parked",
    mobile: "",
    color: "",
    make: "",
    damages: "",
    notes: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    // method to update parent state to trigger adding vehicle and transaction
    addVehicle(formData.ticketNum, "parked", formData.mobile, formData.color, formData.make, formData.damages, formData.notes);
    setFormData(INITIAL_STATE);
    history.push(`/activeGarage`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>Add Vehicle</h3>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
          <Input
            id={val}
            key={`Input-${val}`}
            type={val !== "email" ? "text" : "email"}
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}></Input>
        </FormGroup>
      ))}

      <Button color="primary">Add</Button>
    </Form>
  );
};

export default AddVehicleForm;
