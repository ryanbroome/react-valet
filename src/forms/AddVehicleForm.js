import React, { useState, useContext } from "react";
import { useHistory } from "react-router-dom";
import { FormGroup, Form, Input, Label, Button } from "reactstrap";
import UserContext from "../auth/UserContext";

// add method either passed as props or saved in context ?
const AddVehicleForm = () => {
  const { addVehicle } = useContext(UserContext);
  const history = useHistory();

  const INITIAL_STATE = {
    ticketNum: "",
    mobile: "",
    color: "",
    make: "",
    damages: "",
    notes: "",
  };
  const [formData, setFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((formData) => ({
      ...formData,
      [name]: value,
    }));
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    addVehicle(formData.ticketNum, "parked", formData.mobile, formData.color, formData.make, formData.damages, formData.notes);
    setFormData(INITIAL_STATE);
    history.push(`/transactions/active`);
  };

  return (
    <Form onSubmit={handleSubmit}>
      <h3>Add Vehicle</h3>
      {Object.keys(INITIAL_STATE).map((val, idx) => (
        <FormGroup key={`FormGroup-${val}`}>
          <Input
            id={val}
            key={`Input-${val}`}
            type={val !== "email" ? "text" : "email"}
            name={val}
            placeholder={val.toLowerCase()}
            value={formData[val]}
            onChange={handleChange}
          />
          <Label
            htmlFor={val}
            key={`Label-${idx}`}></Label>
        </FormGroup>
      ))}

      <Button color="primary">Add</Button>
    </Form>
  );
};

export default AddVehicleForm;
