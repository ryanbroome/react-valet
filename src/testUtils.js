import React from "react";
import UserContext from "./auth/UserContext";
// import { render } from "@testing-library/react";
// import { MemoryRouter } from "react-router";
import App from "../src/App";
// import TransactionList from "./TransactionList";

const MockUserContext = ({ children }) => {
  const mockToken = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6InRlc3R1c2VyIiwibG9jYXRpb25JZCI6MSwiaXNBZG1pbiI6ZmFsc2UsImlhdCI6MTcxMzA3NTM4Mn0.dXlnawfzk6wlsgpouoSp58QLvlOOSJFC03iq_FXab-I";
  const mockUserDetail = {
    userId: 1,
    vehicleId: 1,
    locationId: 1,
    transactionId: 1,
    ticketNum: "1",
    mobile: "5553334444",
    color: "red",
    make: "honda",
    username: "testuser",
  };
  const mockLocation = {};
  const mockVehicle = {};
  const mockTransaction = {};

  return <UserContext.Provider value={{ token: mockToken, userDetail: mockUserDetail, location: mockLocation, vehicle: mockVehicle, transaction: mockTransaction, update: App.update }}>{children}</UserContext.Provider>;
};

export { MockUserContext };
