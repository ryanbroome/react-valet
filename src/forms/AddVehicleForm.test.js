import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import AddVehicleForm from "./AddVehicleForm";
import { MemoryRouter } from "react-router";
import { MockUserContext } from "../testUtils";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <MockUserContext children={<AddVehicleForm />}></MockUserContext>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <MockUserContext children={<AddVehicleForm />}></MockUserContext>
    </MemoryRouter>
  );
});
