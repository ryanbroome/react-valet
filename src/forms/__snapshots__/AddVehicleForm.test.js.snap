// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class=""
  >
    <h3>
      Add Vehicle
    </h3>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="ticketNum"
        name="ticketNum"
        placeholder="ticketnum"
        type="text"
        value=""
      />
      <label
        class=""
        for="ticketNum"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="mobile"
        name="mobile"
        placeholder="mobile"
        type="text"
        value=""
      />
      <label
        class=""
        for="mobile"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="color"
        name="color"
        placeholder="color"
        type="text"
        value=""
      />
      <label
        class=""
        for="color"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="make"
        name="make"
        placeholder="make"
        type="text"
        value=""
      />
      <label
        class=""
        for="make"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="damages"
        name="damages"
        placeholder="damages"
        type="text"
        value=""
      />
      <label
        class=""
        for="damages"
      />
    </div>
    <div
      class="position-relative form-group"
    >
      <input
        class="form-control"
        id="notes"
        name="notes"
        placeholder="notes"
        type="text"
        value=""
      />
      <label
        class=""
        for="notes"
      />
    </div>
    <button
      class="btn btn-primary"
    >
      Add
    </button>
  </form>
</DocumentFragment>
`;
